__tests__/utils/bigint.test.js.ts

```diff
@@ -54,18 +54,41 @@ const bigIntObjSchema = z.object({
 
 describe('test JSONBigInt', () => {
   test('should parse numbers', () => {
+    // Doubles should be parsed normally as JS Numbers.
     expect(JSONBigInt.parse('123')).toStrictEqual(123);
     expect(JSONBigInt.parse('123.456')).toStrictEqual(123.456);
     expect(JSONBigInt.parse('1.0')).toStrictEqual(1);
     expect(JSONBigInt.parse('1.000000000000')).toStrictEqual(1);
-    expect(JSONBigInt.parse('12345678901234567890')).toStrictEqual(12345678901234567890n);
-    expect(JSONBigInt.parse('12345678901234567890.000')).toStrictEqual(12345678901234567890n);
```

Comment:
Why did you remove this?

Reply:
The first line was just moved below.

The second line was actually kind of incorrect. Before, I was using `Number.isInteger()` to check whether a parsed JSON value was an integer, and if it returned `true`, I would try the `BigInt` check. Because of that, another problem was created: JS returns `true` for floats ending in `.0`, so `Number.isInteger(123.0)` returns `true`. That's why I also had to add the check for zeroes in the previous code. It was convoluted and unnecessary, because I shouldn't be using `Number.isInteger()` in the first place.

Then, consider the explanation in https://github.com/HathorNetwork/hathor-wallet-lib/pull/805#discussion_r1884282621. The same theory applies in this case. in JS and Python:

```python
>>> json.loads('12345678901234567890.000') == 12345678901234567000.0
True
```

```js
> JSON.parse('12345678901234567890.000') == 12345678901234567000.0
true
```

Summing up, I was unnecessarily trying to parse this number as a `BigInt`, while we should actually just let it be parsed as a double by JS and Python. In practice, this number will never even appear in a JSON generated by these languages.